frontend/src/components/admin/DownloadDataModal.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect } from "react";
+import { useLazyQuery } from "@apollo/client";
+import { Box, Text, Radio, RadioGroup, Stack } from "@chakra-ui/react";
+import { Modal } from "../common/Modal";
+import { GET_USER_COUNT_BY_TOWN } from "../../APIClients/queries/UserQueries";
+import { downloadCSV } from "../../utils/CSVUtils";
+import DatePicker from "./DatePicker";
+
+type DownloadProps = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const DownloadDataModal = ({
+  isOpen,
+  onClose,
+}: DownloadProps): React.ReactElement => {
+  const [radioOption, setRadioOption] = useState("1");
+  const [startMonth, setStartMonth] = useState("");
+  const [startYear, setStartYear] = useState("");
+  const [endMonth, setEndMonth] = useState("");
+  const [endYear, setEndYear] = useState("");
+  const [getCounts, { loading }] = useLazyQuery(GET_USER_COUNT_BY_TOWN);
+  const [downloadEnabled, setDownloadEnabled] = useState(true);
+  const [rangeInvalid, setRangeInvalid] = useState(false);
+
+  useEffect(() => {
+    if (radioOption === "2") {
```

Comment:
Instead of doing this in a useEffect, it would be good to put it directly in the onChange handler for the controls (perhaps refactored into a helper?). That way we can avoid the extra rerender needed to actually update this state.

Reply:
Finding this to be a little tricky to implement. At the moment it's not validating exactly when the input is valid, but rather requires one extra change to the input for the validation to run. I'm guessing this is because updating the state is asynchronous and the validation happens before the state is updated. Any ideas on how to fix this?

Reply:
I guess if it works it's not the end of the world to keep it this way, if the fix is complicated. FF to skip this one.

Reply:
FF 💀